### DISCUSSION TITLE
OAuth flow always redirects to login page

### DISCUSSION THREAD

**Discussion by antoineco** [benbjohnson/wtf#34] [Q&A] [answered]
I'm asking that question here instead of raising a GitHub issue because the problem is most likely on my end.

Whenever I click "Log in with GitHub", I see the following requests being successfully sent:

1. `https://example.com:3000/oauth/github` -> `302`
1. `https://github.com/login/oauth/authorize?client_id=<redacted>&response_type=code&state=<redacted>` -> `302`
1. `https://example.com:3000/?code=<redacted>&state=<redacted>` -> `302`

but then I always get redirected to `https://example.com:3000/login`, as if nothing happened. What am I missing?

**Comment by keybits** [comment, marked as answer]
Did you set the `Authorization callback URL` to `http://localhost:300/oauth/github/callback` when you registered the OAuth application?

It's not explicitly called out in the README. I found the callback path here: https://github.com/benbjohnson/wtf/blob/main/http/auth.go#L23

**Reply by antoineco** [reply]
Ah that was the issue indeed, I didn't even think about looking into the HTTP routes 🤦 
Cheers!

I just sent #35 to clarify that.
